### ISSUE TITLE
Installation fails on Armbian with Kernel 4.17.x on ODROID C2

### ISSUE THREAD

**Issue by Jojo-A** [alexreinert/piVCCU#58]
Hi,

I installed the latest Armbian Testing Image (5.49) with Kernel version 4.17 on my ODROID C2.
I followed the "[Manual Installation](https://github.com/alexreinert/piVCCU/blob/master/docs/setup/armbian.md)" step by step.
As far as I can tell, everything worked until step 8 to finally install pivccu. When installing pivccu, the installation process brings up a box that states:
`WARNING: Could not locate raw uart interface. (...)`. After confirming this, the installation finishes with the not in terminal `Job for pivccu.service failed because the control process exited with error code.`
Executing pivccu-info tells that:
```
Kernel modules: Not available
Raw UART dev: Not available
HMRF Hardware: unknown
(...)
lxc doesn't exist
```
Any idea how I can solve that? Hace you tried to install piVCCU on latest Armbian?

Greetings

**Comment by alexreinert** [comment]
Hi,

the latest armbian image does not contain the kernel headers and the kernel headers in the apt repository of Armbian are not at the latest version so they do not match to the installed kernel.

I'm trying to build a new Odroid C2 image soon, until then the only solution would be a downgrade of the installed kernel, so that it matches the headers of the apt repository.

Alex

**Comment by Jojo-A** [comment]
Hi,
ok, generally I understand. But I thought that the command
```
sudo apt install `dpkg --get-selections | grep 'linux-image-' | grep '\sinstall' | sed -e 's/linux-image-\([a-z0-9-]\+\).*/linux-headers-\1/'`
```
would ensure that the correct headers get installed... Did I misunderstand that?

Greetings

**Comment by alexreinert** [comment]
Yes, but in the wrong version as the kernel in the image is not installed using the apt repository.

**Comment by alexreinert** [comment]
Armbian released new packages for the Odroid C2 (kernel and headers). If you update to them using apt update && apt upgrade piVCCU should work after a reboot (please wait some time after the reboot as the kernel modules are compiled after that reboot and this takes a few minutes)

And I also build a new image for the Odroid C2 using the CCU firmware 2.35.15 (https://www.pivccu.de/images/testing/Armbian_5.50_Odroidc2_Ubuntu_xenial_next_piVCCU_2018-06-29.zip)

**Comment by Jojo-A** [comment]
Sadly the instructions above did not work with my current installation of Armbian. The headers where updated, piVCCU kernel modules became "Available", but the raw UART interface was still not recognized, even after a complete uninstall/reinstall if piVCCU.
I will now try the image from your link. I will report back my results.
Thank you!

**Comment by alexreinert** [comment]
Thanks for the feedback.

I found the issue, why your last manual installation did not work:
With the new armbian kernel there was a little change in the device tree. I already fixed this some days ago, but the deb package was only in the testing apt repo and not in the stable one. As the new image was build using the testing repo, the image works without problems.
I now copied the new device tree package to the stable apt repo, too. So the issue should be fixed now for the manual installation, too.
